build_tag_push.py

```diff
@@ -1,101 +1,130 @@
-	# in my parseAddFile function, I am indexing that list to get the name out.
-	os.system("echo $IMAGE_NAME") # This is just a check to make sure I am getting the correct result. Success.
-	os.system("docker build -t $IMAGE_NAME .") 
-	# This . means that you are in the current directory holding the Dockerfile that you need.
-	# You can give an exact location instead of the . so for my setup it would be /Users/alyssakelley/Desktop/MVP Studio/DockerFiles
-	# You can have the -t flag to tag the image. So it would be: docker build -t imagename:tag . 
-	# Note: Then if you want to run that Docker image that was just created, you can do the following command in the terminal:
-	# 	docker run <image ID> 
-	# Note: The Image ID you can access with the command 'docker images'
-	image_id = os.system("docker images -q $IMAGE_NAME")
-	print image_id
-	#os.environ['IMAGE_ID'] = image_id  # This -q flag retrieves the last image ID (Doesn't work... TypeError)
-	#os.system("docker run $IMAGE_ID")
-	#os.system("docker run docker images -q $IMAGE_NAME") -> Nope.
-	#os.system("docker images -q $IMAGE_NAME")
-	#os.system("IMAGE_ID=$(docker images -q $IMAGE_NAME)")
-	os.system("echo done")
-	#os.system('docker run $IMAGE_ID')
-
-	##########################################
-	# NOT FINISHED HERE -- WILL PICK UP ON IT.
-	# Left off on trying to store the Image ID 
-	# as an environment variable to use that when 
-	# running the ID. 
-	##########################################
-
-
-# Step 4 - Tags the image with the current commit hash & pushed to Docker Hub = NOT COMPLETE.
-#
-#
-#
-#
-#
-#
-#
-#
-#
-#
-#
+
+def build_DockerImage(filename):
+	docker_dir_context = createContextDirectory(parseAppFile(filename)['docker_context']) # This will be the build and static file (and needs a Dockerfile as well)
+	image_name = parseAppFile(filename)['name'] # Name from app.yml file
+	subprocess.check_call(["docker", "build", docker_dir_context, "-t", image_name]) # this builds the docker image.
+
+	# QUESTION: When I try to build the image from the file that has the docker context from the app.yml file 
+	# then I get the error: "Build an image from a Dockerfile". The only way I was able to complete this was 
+	# putting in my own Dockerfile, which I am assuming the groups will have as well and we will need to copy
+	# this over into that directory created, or is this going to be copied over manually?
+
+# Step 4 - Tags the image with the current commit hash & push to Docker Hub = COMPLETE.
+
+def tag_DockerImage(filename):
+	'''
+	This function is finding the tagging the Docker Image with the name of the user/image_name, and the current git hash. 
+	Note: The current git hash is shortened a bit, but to display the full hash, then change the lower case %h to upper case %H. 
+	The source for this command is: https://container-solutions.com/tagging-docker-images-the-right-way
+	'''
+	image_name = parseAppFile(filename)['name']
```

Comment:
you should probably parse the `app.yml` once in main and then pass the resulting dict from method to method instead of re-parsing it over and over.